p2p/src/network/pubsub/p2p_network_pubsub_reducer.rs

```diff
@@ -687,6 +672,31 @@ impl P2pNetworkPubsubState {
                 Ok(())
             }
             P2pNetworkPubsubAction::IgnoreMessage { .. } => Ok(()),
+            P2pNetworkPubsubAction::BroadcastMessage { message_id } => {
+                let Some(message) = pubsub_state.mcache.map.get(&message_id) else {
+                    bug_condition!("Message with id: {:?} not found", message_id);
+                    return Ok(());
+                };
+
+                let raw_message = message.message().clone();
+                let peer_id = *message.peer_id();
+
+                pubsub_state.reduce_incoming_validated_message(message_id, peer_id, &raw_message);
```

Comment:
`fn reduce_incoming_validated_message` would benefit from a comment explaining on a high level what it does (similar to `reduce_incoming_message` but doesn't need to be as extense). Could be this:

```
/// Queues a validated message for propagation to other peers in the pubsub network.
/// For peers that are "on mesh" for the message's topic, queues the full message.
/// For other peers, queues an IHAVE control message to notify about message availability.
/// The original sender is excluded from propagation.
```